application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -474,45 +474,70 @@ public function set_survey_properties($sSessionKey, $iSurveyID, $aSurveyData)
      * @return array in case of success result of the activation
      */
-    public function activate_survey($sSessionKey, $iSurveyID)
+    public function activate_survey($sSessionKey, $iSurveyID, $activationSettings = [])
     {
-        if ($this->_checkSessionKey($sSessionKey)) {
-            $iSurveyID = (int) $iSurveyID;
-            $oSurvey = Survey::model()->findByPk($iSurveyID);
-            if (is_null($oSurvey)) {
-                return array('status' => 'Error: Invalid survey ID');
-            }
-            if ($oSurvey->isActive) {
-                return array('status' => 'Error: Survey already active');
-            }
-            // Check consistency for groups and questions
-            Yii::app()->loadHelper('admin/activate');
-            $checkHasGroup = checkHasGroup($iSurveyID);
-            $checkGroup = checkGroup($iSurveyID);
-
-            if ($checkHasGroup !== false || $checkGroup !== false) {
-                return array('status' => 'Error: Survey does not pass consistency check');
-            }
-
-            $surveyActivator = new SurveyActivator($oSurvey);
-
-
-            if (Permission::model()->hasSurveyPermission($iSurveyID, 'surveyactivation', 'update')) {
-                $aActivateResults = $surveyActivator->activate();
-
-                if (isset($aActivateResults['error'])) {
-                    return array('status' => 'Error: ' . $aActivateResults['error']);
-                } else {
-                    return $aActivateResults;
-                }
-            } else {
-                            return array('status' => 'No permission');
-            }
-        } else {
-                    return array('status' => self::INVALID_SESSION_KEY);
+        // check provided session key and setup user session
+        if (!$this->_checkSessionKey($sSessionKey)) {
+            return array('status' => self::INVALID_SESSION_KEY);
+        }
+        $iSurveyID = (int) $iSurveyID;
+        // check valid survey id and current user has permission
+        if (!Permission::model()->hasSurveyPermission($iSurveyID, 'surveyactivation', 'update')) {
+            return array('status' => 'No permission');
         }
+        $oSurvey = Survey::model()->findByPk($iSurveyID);
+        // check if survey is already activated
+        if ($oSurvey->isActive) {
+            return array('status' => 'Error: Survey already active');
+        }
+        // Check consistency for groups and questions
+        Yii::app()->loadHelper('admin/activate');
+        $checkHasGroup = checkHasGroup($iSurveyID);
+        $checkGroup = checkGroup($iSurveyID);
+        if ($checkHasGroup !== false || $checkGroup !== false) {
+            return array('status' => 'Error: Survey does not pass consistency check');
+        }
+        $surveyActivator = new SurveyActivator($oSurvey);
+        // grab computed survey settings from common helper
+        $surveysettings = getSurveyInfo($iSurveyID);
```

Comment:
My opinion : we must remove usage of getSurveyInfo if we can avoid : it create a big data with a lot of unused value.

Reply:
Yes, I can refactor to remove this